nomad/job_endpoint_hooks.go

```diff
@@ -302,6 +305,14 @@ func (jobImpliedConstraints) Mutate(j *structs.Job) (*structs.Job, []error, erro
 			mutateConstraint(constraintMatcherLeft, tg, vaultConstraintFn(vaultBlock))
 		}
 
+		secretTasks := lang.MapKeys(secretBlocks[tg.Name])
+		for _, secretTask := range secretTasks {
+			secretBlock := secretBlocks[tg.Name][secretTask]
+			for _, secret := range secretBlock {
+				mutateConstraint(constraintMatcherLeft, tg, secretsConstraintFn(secret))
+			}
+		}
```

Comment:
The `j.Secrets()` method produces a map of job -> group -> task, so if we have multiple tasks with the same secret provider (or even multiple secrets within the same task), don't we end up with multiple identical constraint blocks defined in a task group?

For purposes of the constraint, I think we can flatten that whole map down to a `map[string][]string` (a map of group names to the list of providers). That's not what's being done for `j.Vault()`, but I think that method is as complicated as it is because it's used elsewhere too.